[Content] Clarify use cases for buttons and links (#2144)
Shopify/polaris-react: src/components/Link/README.md

# Link

Links take users to another place, and usually appear within or directly following a sentence.

> For actions that aren't related to navigation, use the [button component](/components/actions/button).

Reviewer: ```suggestion
For actions that aren’t related to navigation, use the [button component](https://polaris.shopify.com/components/actions/button).
```

---

## Best practices

### Buttons versus links

Links are used primarily for navigation, and usually appear within or directly following a sentence.

> Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as "view shipping settings".

Reviewer: ```suggestion
Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as “view shipping settings”.
```

> The HTML for buttons and links carries specific meaning. Using these components intentionally and consistently results in:

Reviewer: Making the first part of this sentence clearly refer to React components could clarify the intent:

```suggestion
The markup resulting from `Button` and `Link` carries meaning. Using these components intentionally and consistently results in:
```

What do you think?

- a more inclusive experience for assistive technology users
- a more cohesive visual experience for sighted users
- products that are easier to maintain at scale

### Unstyled links

> If the existing link styles don't meet the needs of your project, then use the `UnstyledLink` component to create a custom link style.

Reviewer: ```suggestion
If the existing link styles don’t meet the needs of your project, then use the `UnstyledLink` component to create a custom link style.
```

---

## Content guidelines

> The link component should follow the content guidelines for [links](/content/actionable-language#section-links).

Reviewer: ```suggestion
The link component should follow the content guidelines for [links](https://polaris.shopify.com/content/actionable-language#section-links).
```

## Related components

> - For actions that don't appear within or directly following a sentence, use the use the [button component](/components/actions/button)

Reviewer: ```suggestion
- For actions that don’t appear within or directly following a sentence, use the [button component](https://polaris.shopify.com/components/actions/button)
```
